module/input/NatNet/src/NatNet.cpp

```diff
@@ -402,10 +422,15 @@ namespace module::input {
 
 
     void NatNet::process(const std::vector<char>& input) {
-
         // Get our packet
         const Packet& packet = *reinterpret_cast<const Packet*>(input.data());
 
+        // std::stringstream ss;
+        // for (const auto& c : input) {
+        //     ss << std::setw(2) << std::setfill('0') << std::hex << int(c);
+        // }
+        // NUClear::log(ss.str());
```

Comment:
Something like this might work for dumping packets out (with an appropriate config value)
Note this code has not been tested!
You also may want to add a config value for where to dump the packets to so you don't fill up your folder

```suggestion
        if (config.dump_packets) {
            static int i = 0;
            std::ofstream output(std::format("natnet_{:06d}.bin", i++));
            output.write(input.data(), input.size());
        }
```